x-pack/plugins/security_solution/server/assistant/tools/esql_language_knowledge_base/esql_language_knowledge_base_tool.ts

```diff
@@ -21,30 +20,29 @@ const toolDetails = {
 export const ESQL_KNOWLEDGE_BASE_TOOL: AssistantTool = {
   ...toolDetails,
   sourceRegister: APP_UI_ID,
-  isSupported: (params: AssistantToolParams): params is EsqlKnowledgeBaseToolParams => {
-    const { chain, isEnabledKnowledgeBase, modelExists } = params;
-    return isEnabledKnowledgeBase && modelExists && chain != null;
+  isSupported: (params: AssistantToolParams): params is AssistantToolParams => {
+    const { kbDataClient, isEnabledKnowledgeBase, modelExists } = params;
+    return isEnabledKnowledgeBase && modelExists && kbDataClient != null;
   },
   getTool(params: AssistantToolParams) {
     if (!this.isSupported(params)) return null;
 
-    const { chain } = params as EsqlKnowledgeBaseToolParams;
-    if (chain == null) return null;
+    const { kbDataClient } = params as AssistantToolParams;
+    if (kbDataClient == null) return null;
 
     return new DynamicStructuredTool({
       name: toolDetails.name,
       description: toolDetails.description,
       schema: z.object({
         question: z.string().describe(`The user's exact question about ESQL`),
       }),
-      func: async (input, _, cbManager) => {
-        const result = await chain.invoke(
-          {
-            query: input.question,
-          },
-          cbManager
-        );
-        return result.text;
+      func: async (input) => {
+        const docs = await kbDataClient.getKnowledgeBaseDocumentEntries({
+          kbResource: ESQL_RESOURCE,
+          query: input.question,
+        });
```

Comment:
So I think we're gonna have to add a method to the `kbDataClient` that mirrors the old behavior of `similaritySearch` in esStore, as this isn't always going to include the `required:true` example queries that the previous implementation did. This query will also not work for the `kbResource:'unknown'` documents from `8.15` I think (the previous `esStore.similaritySearch()` didn't filter on that value for the non-required docs).

Reply:
Changes in https://github.com/elastic/kibana/pull/184885/commits/9c60dca7e6dd8bc50666bc83cf4cf5a460d16815 LGTM 👍